src/python/pants/backend/tools/trufflehog/subsystem.py

```diff
@@ -0,0 +1,76 @@
+# Copyright 2024 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+"""Downloads the trufflehog binary."""
+
+from __future__ import annotations
+
+from pants.core.util_rules.config_files import ConfigFilesRequest
+from pants.core.util_rules.external_tool import ExternalTool
+from pants.engine.platform import Platform
+from pants.option.option_types import BoolOption, SkipOption, StrListOption
+from pants.util.strutil import softwrap
+
+
+class Trufflehog(ExternalTool):
+    """A tool for finding secrets in code."""
+
+    options_scope = "trufflehog"
+    name = "Trufflehog"
+    default_version = "v3.34.0"
+    help = "Trufflehog secrets scanning"
+    skip = SkipOption("lint")
+    macos_arm_sha = "19e10e34e95d797cbb924b342b873caa9e71296f9bead28b390d96981f47fbb0"
```

Comment:
I'd recommend inlining these into the default_known_versions, because these are just one set of known versions - we start out with just `v3.34.0`, but in the future we will presumably add more known trufflehog versions, so `default_known_versions` might have 8, or 12 or any multiple of 4 entries of known versions, and Pants then selects one based on your platform and the version you request. So there is nothing special about these shas that indicates that they should have their own constants. Does that make sense?

Reply:
Inlined them now.